hdf5-derive/src/lib.rs

```diff
@@ -176,9 +175,8 @@ fn impl_trait(
                     .filter(|&(_, f)| !is_phantom_data(&f.ty))
                     .map(|(i, f)| (Index::from(i), f))
                     .unzip();
-                if fields.is_empty() {
-                    panic!("Cannot derive H5Type for empty tuple structs");
-                }
+                assert!(!fields.is_empty(), "Cannot derive H5Type for empty tuple structs");
```

Comment:
Todo note to self: we should really use `#[proc_macro_error]` which will emit nice errors attached to spans (it's just when hdf5-derive was written, proc-macro-error wasn't there).